### PR TITLE
Feature/#768 increase okewo integration to sb3

### LINKED ISSUE #768: Migrate digiwf-integrations to Spring Boot 3
Migrate digiwf-integrations to Spring Boot 3 (see [Migration Guide](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide)).

**Todo**

- Upgrade Spring Boot
- Upgrade Spring Security
- Configure Properties
	- Use the spring-boot-properties-migrator module to detect props that changed
- Rename `spring.factories` to `META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports`
- Upgrade Dependencies
- Open a PR for each integration
- Ignore Legacy Integrations

Integrations: 
- [x] address integration --> Stephan
- [x] alw integration --> Stephan
- [x] cosys integration --> Julia
- [x] dms integration --> Julia
- [x] email integration --> Julia
- [x] example integration --> Stephan
- [ ] okewo-integration  --> Stephan
- [x] s3-integration --> Simon Z
- [x] verification integration remove from repo --> Stephan

**Additional Context**

- REST Template https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#apache-httpclient-in-resttemplate
- Flyway https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#flyway

### PR DESCRIPTION
**Pull request by StephanStrehlerCGI**
### Description

Short description or comments

### Reference

Issues: closes #768

### Check-List

- [x] All Acceptance criteria of user story are met
- [ ] JUnit tests are written (60% CodeCov)
- [ ] [Internal Review]([https://github.com/it-at-m/digiwf-core/blob/dev/CHANGELOG.md](https://confluence.muenchen.de/display/MPdZ/Review+-+DigiWF)) is maintained
- [ ] Documentations [external](https://github.com/it-at-m/digiwf-core/tree/dev/docs) and [internal](https://wiki.muenchen.de/betriebshandbuch/index.php?title=DigiWF&sfr=betriebshandbuch) are completed
- [x] Smoketest successful (Manual E2E-Test depending on Change) 
- [x] No Branch waste left
- [ ] [Board](https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/board) is up-to-date
- [ ] Openshift environments are prepared (Secrets, etc.) and release-issue is maintained

### CODE REVIEW

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    <parent>
+        <groupId>de.muenchen.oss.digiwf</groupId>
+        <artifactId>digiwf-okewo-integration</artifactId>
+        <version>1.3.0-SNAPSHOT</version>
+    </parent>
+
+    <properties>
+        <apache.commons.version>4.4</apache.commons.version>
+        <swagger.codegen.version>3.0.33</swagger.codegen.version>
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml]: why is it needed?

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    </properties>
+
+    <dependencies>
+        <!-- Spring -->
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml]: drop it

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        <dependency>
+            <groupId>org.openapitools</groupId>
+            <artifactId>jackson-databind-nullable</artifactId>
+            <version>0.2.6</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml:39]: can be removed (is a part of default Maven plugin included into build)

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>${maven.surefire.plugin.version}</version>
+            </plugin>
+            <plugin>
+                <groupId>org.openapitools</groupId>
+                <artifactId>openapi-generator-maven-plugin</artifactId>
+                <version>7.0.1</version>
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml]: the version is managed, should not be here.

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                <executions>
+                    <execution>
+                        <id>generate-server-stubs</id>
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml]: the execution is probably called badly, we are generating client code here

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+                            <apiPackage>de.muenchen.oss.digiwf.okewo.integration.client.api</apiPackage>
+                            <modelPackage>de.muenchen.oss.digiwf.okewo.integration.client.model</modelPackage>
+                            <invokerPackage>de.muenchen.oss.digiwf.okewo.integration.client</invokerPackage>
+
+                            <generateApiTests>false</generateApiTests>
+                            <generateModelTests>false</generateModelTests>
+                            <generateApiDocumentation>false</generateApiDocumentation>
+                            <generateModelDocumentation>false</generateModelDocumentation>
+                            <generateSupportingFiles>false</generateSupportingFiles>
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml]: if generate supporting files is set to false, why do we need to specify supporting files below? usually we use a .openapi-generator-ignore to exclude stuff.

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+                                <java8>true</java8>
+                                <hideGenerationTimestamp>true</hideGenerationTimestamp>
+                                <sourceFolder>src/main/java/</sourceFolder>
+                                <useJakartaEe>true</useJakartaEe>
+                            </configOptions>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml]: default plugin same as surefire, no need to list it

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-core/pom.xml [hunk 8]
```diff
@@ -17,31 +17,29 @@
...
 
-        <!-- Validation -->
-        <dependency>
-            <groupId>javax.validation</groupId>
-            <artifactId>validation-api</artifactId>
-        </dependency>
-
         <!-- openApi -->
         <dependency>
             <groupId>org.springdoc</groupId>
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-core/pom.xml]: this dependency is called differently in SPB3

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-core/src/main/resources/application.yml [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+spring:
+  application:
+    name: DigiWF OK.EWO Integration
+  banner:
+    location: banner.txt
+  lifecycle:
+    timeout-per-shutdown-phase: 1m
+  cloud:
+    function:
+      definition: functionRouter;sendMessage;
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-core/src/main/resources/application.yml]: routing-expression: "headers['type']" missing?

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-service/pom.xml [hunk 10]
```diff
@@ -117,6 +90,17 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.springframework.kafka</groupId>
+            <artifactId>spring-kafka-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.wiremock</groupId>
+            <artifactId>wiremock-standalone</artifactId>
+            <version>3.3.1</version>
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-service/pom.xml:94]: is managed, remove the version

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-core/src/main/resources/application.yml [hunk 11]
```diff
@@ -0,0 +1,71 @@
+management:
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-core/src/main/resources/application.yml]: why this file is present in the core? it is slightly confusing... I believe it should be in service only. If you need it for the itest setup, please merge it with the application-itest.yaml

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-service/src/main/resources/application.yml [hunk 12]
```diff
@@ -30,3 +30,72 @@ spring:
     location: banner.txt
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-service/src/main/resources/application.yml]: default

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-service/src/main/resources/application.yml [hunk 13]
```diff
@@ -30,3 +30,72 @@ spring:
...
+      message:
+        incidentDestination: "dwf-connector-incident-${DIGIWF_ENV}"
+        bpmnErrorDestination: "dwf-connector-bpmnerror-${DIGIWF_ENV}"
+        correlateMessageDestination: "dwf-connector-${DIGIWF_ENV}"
+        typeMappings:
+          getPerson: "getPerson"
+          searchPerson: "searchPerson"
+          getPersonErweitert: "getPersonErweitert"
+          searchPersonErweitert: "searchPersonErweitert"
+# FIXME: move to io.muenchendigital
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-service/src/main/resources/application.yml]: why not doing it right now?

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-service/src/test/resources/application-itest.yml [hunk 14]
```diff
@@ -4,6 +4,103 @@ de:
       digiwf:
         okewo:
           password: 'password'
-          url: 'url'
+          url: 'http://localhost:8089'
           username: 'username'
           benutzerId: 'benutzerId'
+
+
+management:
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-service/src/test/resources/application-itest.yml]: application-itest should only overwrite relevant properties. no reason to duplicate the entire file

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-starter/src/main/java/de/muenchen/oss/digiwf/okewo/integration/configuration/OkEwoIntegrationAutoConfiguration.java [hunk 15]
```diff
@@ -1,93 +1,109 @@
...
+    excludeFilters = {
+        @ComponentScan.Filter(
+            type = FilterType.ASSIGNABLE_TYPE,
+            classes = {
+                /**
+                 * Exclude to avoid multiple instantiation of beans with same name.
+                 * This class is instantiated in {@link OkEwoIntegrationAutoConfiguration}
+                 * to give the bean another name.
+                 */
+                ApiClient.class,
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-starter/src/main/java/de/muenchen/oss/digiwf/okewo/integration/configuration/OkEwoIntegrationAutoConfiguration.java]: this is strange and should not be needed. you may component scan the same component multiple time. why is this needed?

File: digiwf-integrations/digiwf-okewo-integration/pom.xml [hunk 16]
```diff
@@ -18,11 +18,16 @@
     </parent>
 
     <properties>
+        <!--todo: remove after the successfully spring boot version upgrade of all migrations-->
+        <spring-boot.version>3.1.5</spring-boot.version>
+        <spring.cloud.version>2022.0.4</spring.cloud.version>
+
         <maven.surefire.plugin.version>3.2.2</maven.surefire.plugin.version>
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/pom.xml]: remove the version

File: digiwf-integrations/digiwf-okewo-integration/pom.xml [hunk 17]
```diff
@@ -61,6 +67,20 @@
...
+
+        <!-- TODO could be removed after migrate all migrations to spring 3-->
+        <dependency>
+            <groupId>de.muenchen.oss.digiwf</groupId>
+            <artifactId>digiwf-message-starter</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
+        <!-- Logging -->
+        <dependency>
```
Comment (zambrovski) [inline digiwf-integrations/digiwf-okewo-integration/pom.xml:77]: renamed - it is now called `digiwf-spring-logging-and-tracing`

**Review by zambrovski — CHANGES_REQUESTED** [review]
Please review the requested changes.

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml (hunk 3 above)
Comment (StephanStrehlerCGI) [reply digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-client/pom.xml:39]: I don't know exactly what you mean. For it looks exacly like the other configurations in task-service or other integrations.

File: digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-service/pom.xml (hunk 10 above)
Comment (StephanStrehlerCGI) [reply digiwf-integrations/digiwf-okewo-integration/digiwf-okewo-integration-service/pom.xml:94]: For that we have to merge your draft branch first

**Review by zambrovski — APPROVED** [review]

**Pull request merged**